### PR TITLE
[0.9.x][UF-364] uberfire-security: add support for WildFly10/EAP7

### PR DESCRIPTION
**Pull request by psiroky**
- both WildFly8/EAP6 and WildFly10/EAP7 are now supported.
  They need to be in separate modules, because the WildFly
  libs in use are not fully backwards compatible (API wise)
- the -wildfly-commons module contains shared code to avoid
  duplication

@romartin please take a look.

### CODE REVIEW

File: uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (romartin) [inline uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml:1]: Is this xml really necessary now? or just a wrong git add? :)

Comment (manstis) [inline uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml:1]: au contraire.. isn't this "expected prolog" http://www.w3schools.com/xml/xml_syntax.asp

Comment (romartin) [inline uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml:1]: @manstis not following you sorry... :/ Prolog is fine isn't it? I was not sure if the whole file was really necessary....  ( sorry it's late and I'm sometimes confused... only asking :)

**Comment by romartin** [comment]
@psiroky btw quick q - I understand this stuff goes to master as well right?

File: uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml (hunk 1 above)
Comment (manstis) [inline uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml:1]: Ah, OK @romartin I thought you was just asking about the prolog... 

@manstis Step away from this PR.. move along... stop being a bother.

Comment (psiroky) [inline uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml:1]: I added the logback config on purpose. By default (without any config) logback uses DEBUG level which pollutes the build output quite a lot.

**Comment by psiroky** [comment]
This should go only into 0.9.x (KIE 6.5.x). On master we support WildFly10+.

File: uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml (hunk 1 above)
Comment (romartin) [inline uberfire-security/uberfire-security-management/uberfire-security-management-wildfly/uberfire-security-management-wildfly10/src/test/resources/logback-test.xml:1]: @psiroky yeahh just saw the debug messages you're right sorry! are good np :)
@manstis hehe np! It's good having you're eyes always looking everywhere :)

**Comment by romartin** [comment]
@psiroky ah ok, but I thnk makes sense moving the new maven structure fro wildfly modules although adding only support for the wf10 one, so when updating to wf11 or whatever we already will have those base module, well only asking not sure yet what will happen on the future with all those modules.. .:)

**Comment by romartin** [comment]
Great work Petr, I see docs updated, the new iface and code looks good, all tests and webapp running.

+1 

Moving to test this one now -> https://github.com/droolsjbpm/kie-wb-distributions/pull/315

**Comment by psiroky** [comment]
I am still not sure if we should do something similar on master. This change brings one more level of indirection (e.g. introducing more abstract classes), which makes it slightly harder to understand (but better than the code duplication there was before). Maybe we could postpone this change on master till it is absolutely needed? WDYT @ederign?

**Comment by psiroky** [comment]
Note: in order to support WildFly 10 (which is Java 8+), it is needed to bump the minimal JDK version to 8 (done in the top-level pom.xml).

**Comment by ederign** [comment]
@psiroky agree with you to keep that just  here (at least for now).

We are ready to merge?

**Comment by psiroky** [comment]
@ederign I am waiting for +1 on https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/244 and https://github.com/droolsjbpm/kie-wb-distributions/pull/315. They need to be merged together with this one.

**Comment by romartin** [comment]
@psiroky @ederign sorry guys, yesterday had to move to another issue, just building https://github.com/droolsjbpm/kie-wb-distributions/pull/315 this one now.

**Comment by romartin** [comment]
@psiroky @ederign verified this one -> https://github.com/droolsjbpm/kie-wb-distributions/pull/315 Thanx!

**Comment by psiroky** [comment]
This is almost ready to merge, I just need one more +1 for https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/244

**Comment by psiroky** [comment]
@csadilek looks like we are ready to go. Could you please merge the PRs in this series?
https://github.com/uberfire/uberfire-extensions/pull/284
https://github.com/droolsjbpm/droolsjbpm-build-bootstrap/pull/244
https://github.com/droolsjbpm/kie-wb-distributions/pull/315

**Pull request merged**